src/python/nimbusml/linear_model/linearsvmbinaryclassifier.py

```diff
@@ -0,0 +1,183 @@
+
+    :param number_of_iterations: Number of iterations.
+
+    :param initial_weights_diameter: Sets the initial weights diameter that
+        specifies the range from which values are drawn for the initial
+        weights. These weights are initialized randomly from within this range.
+        For example, if the diameter is specified to be ``d``, then the weights
+        are uniformly distributed between ``-d/2`` and ``d/2``. The default
+        value is ``0``, which specifies that all the  weights are set to zero.
+
+    :param no_bias: No bias.
+
+    :param initial_weights: Initial Weights and bias, comma-separated.
+
+    :param shuffle: Whether to shuffle for each training iteration.
+
+    :param batch_size: Batch size.
+
+    :param params: Additional arguments sent to compute engine.
+
+    .. index:: models, classification, svm
+
+    Example:
+       .. literalinclude:: /../nimbusml/examples/LinearSvmBinaryClassifier.py
+               :language: python
+    """
+
+    @trace
+    def __init__(
+            self,
+            normalize='Auto',
+            caching='Auto',
+            lambda_=0.001,
+            perform_projection=False,
+            number_of_iterations=1,
+            initial_weights_diameter=0.0,
+            no_bias=False,
+            initial_weights=None,
+            shuffle=True,
+            batch_size=1,
+            feature=None,
+            label=None,
+            weight=None,
+            **params):
+
+        if 'feature_column_name' in params:
+            raise NameError(
+                "'feature_column_name' must be renamed to 'feature'")
+        if feature:
+            params['feature_column_name'] = feature
+        if 'label_column_name' in params:
+            raise NameError(
+                "'label_column_name' must be renamed to 'label'")
+        if label:
+            params['label_column_name'] = label
+        if 'example_weight_column_name' in params:
+            raise NameError(
+                "'example_weight_column_name' must be renamed to 'weight'")
+        if weight:
+            params['example_weight_column_name'] = weight
+        BasePredictor.__init__(self, type='classifier', **params)
+        core.__init__(
+            self,
+            normalize=normalize,
+            caching=caching,
+            lambda_=lambda_,
+            perform_projection=perform_projection,
+            number_of_iterations=number_of_iterations,
+            initial_weights_diameter=initial_weights_diameter,
+            no_bias=no_bias,
+            initial_weights=initial_weights,
+            shuffle=shuffle,
+            batch_size=batch_size,
+            **params)
+        self.feature = feature
+        self.label = label
+        self.weight = weight
+
+    @trace
+    def predict_proba(self, X, **params):
+        '''
+        Returns probabilities
+        '''
+        return self._predict_proba(X, **params)
+
+    @trace
+    def decision_function(self, X, **params):
```

Comment:
>decision_function [](start = 8, length = 17)

Pls add tests for predict_proba() and decision_function()

Reply:
Done